src/helpers/http/mod.rs

```diff
@@ -103,42 +148,36 @@ mod e2e_tests {
         PublicKey::from(decoded_arr)
     }
 
+    // randomly grabs open port
+    fn open_port() -> u16 {
```

Comment:
this is a way to learn a port that is open. Since I can't pass in an open socket directly, I thought this might be the next best thing. Otherwise I have to worry about port collisions across tests

Reply:
probably fine for now, but I would love to change it later. Peer discovery for tests should start with spinning up web servers, learning their ports and then providing peer discovery